node/core/backing/src/lib.rs

```diff
@@ -733,18 +736,16 @@ impl CandidateBackingJob {
 
 	// Make a `PoV` available.
 	//
-	// This calls an inspection function before making the PoV available for any last checks
-	// that need to be done. If the inspection function returns an error, this function returns
-	// early without making the PoV available.
-	#[tracing::instrument(level = "trace", skip(self, pov, with_commitments), fields(subsystem = LOG_TARGET))]
-	async fn make_pov_available<T, E>(
+	// This will compute the erasure root internally and compare it to the expected erasure root.
+	// This returns `Err()` iff there is an internal error. Otherwise, it returns either `Ok(Ok(()))` or `Ok(Err(_))`.
+	#[tracing::instrument(level = "trace", skip(self, pov), fields(subsystem = LOG_TARGET))]
+	async fn make_pov_available(
```

Comment:
It might be worth factoring this function into two: one which checks the erasure root, and one which stores the available data. The goal is not really to reduce the amount of work done; it's more to avoid non-sequitur lines like `let erasure_valid = self.make_pov_available()`.

Reply:
It may help remove this pesky nested `Result` as well

Reply:
beyond the scope of this PR